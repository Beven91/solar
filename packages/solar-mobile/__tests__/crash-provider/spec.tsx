import React from 'react';
import CrashProvider from '../../src/crash-provider';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

class ThrowError extends React.Component<any, any> {
  render() {
    const { error } = this.props;
    if (error) {
      throw new Error('MOCKERROR '+error);
    }
    return (
      <div>
        hello.....
      </div>
    );
  }
}

describe('Crash', () => {
  it('renders normal', () => {
    const tree = mount(
      <CrashProvider>
        <div>胜多负少防守打法</div>
      </CrashProvider>
    );
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders throw error', () => {
    const tree = mount(
      <CrashProvider>
        <ThrowError error="渲染失败...." />
      </CrashProvider>
    );
    expect(toJson(tree)).toMatchSnapshot();
  });


  it('renders error and custom Error', () => {
    class MyError extends React.Component {
      render() {
        return <div>AAA</div>;
      }
    }
    const tree = mount(
      <CrashProvider view={(props)=><MyError {...props} />}>
        <ThrowError error="渲染失败...." />
      </CrashProvider>
    );
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders error control', () => {
    const tree = mount(
      <CrashProvider>
        <div>hello</div>
      </CrashProvider>
    );
    expect(toJson(tree)).toMatchSnapshot();

    tree.setProps({ visible: true });
    expect(toJson(tree)).toMatchSnapshot();
  });
});
