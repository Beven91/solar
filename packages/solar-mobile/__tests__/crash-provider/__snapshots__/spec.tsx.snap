// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crash renders error and custom Error 1`] = `
<CrashProvider
  view={[Function]}
  visible={false}
>
  <MyError
    error="渲染异常"
  >
    <div>
      AAA
    </div>
  </MyError>
</CrashProvider>
`;

exports[`Crash renders error control 1`] = `
<CrashProvider
  view={[Function]}
  visible={false}
>
  <div>
    hello
  </div>
</CrashProvider>
`;

exports[`Crash renders error control 2`] = `
<CrashProvider
  view={[Function]}
  visible={true}
>
  <Crash
    btnText="返回"
    className=""
    icon=""
    message="大侠别急，您可点击以返回试试~"
    onClick={[Function]}
    title="页面竟然崩溃了"
  >
    <div
      className="crash-component  "
    >
      <img
        className="logo"
        src=""
      />
      <div
        className="crash-title"
      >
         
        页面竟然崩溃了
      </div>
      <div
        className="crash-description"
      >
        大侠别急，您可点击以返回试试~
      </div>
      <div
        className="crash-button"
        onClick={[Function]}
      >
         
        返回
         
      </div>
    </div>
  </Crash>
</CrashProvider>
`;

exports[`Crash renders normal 1`] = `
<CrashProvider
  view={[Function]}
  visible={false}
>
  <div>
    胜多负少防守打法
  </div>
</CrashProvider>
`;

exports[`Crash renders throw error 1`] = `
<CrashProvider
  view={[Function]}
  visible={false}
>
  <Crash
    btnText="返回"
    className=""
    icon=""
    message="大侠别急，您可点击以返回试试~"
    onClick={[Function]}
    title="页面竟然崩溃了"
  >
    <div
      className="crash-component  "
    >
      <img
        className="logo"
        src=""
      />
      <div
        className="crash-title"
      >
         
        页面竟然崩溃了
      </div>
      <div
        className="crash-description"
      >
        大侠别急，您可点击以返回试试~
      </div>
      <div
        className="crash-button"
        onClick={[Function]}
      >
         
        返回
         
      </div>
    </div>
  </Crash>
</CrashProvider>
`;
